src/database/columns.ts

```diff
@@ -0,0 +1,46 @@
+import { AgendaItem } from '@/database/allDbTypes';
+
+type Assert<_T extends true> = never;
+type Equivalent<A extends string, B extends string> = A extends B
+  ? B extends A
+    ? true
+    : `${A} does not match ${B}`
+  : `${B} does not match ${A}`;
+type ColumnsMatch<A extends object, B extends readonly string[]> = Equivalent<
+  keyof A & string,
+  B[number]
+>;
+
+export const agendaItemColumns = [
+  'address',
+  'agendaCd',
+  'agendaItemAddress',
+  'agendaItemId',
+  'agendaItemRecommendation',
+  'agendaItemSummary',
+  'agendaItemTitle',
+  'backgroundAttachmentId',
+  'councilAgendaItemId',
+  'decisionAdvice',
+  'decisionBodyId',
+  'decisionBodyName',
+  'decisionRecommendations',
+  'geoLocation',
+  'id',
+  'itemProcessId',
+  'itemStatus',
+  'meetingDate',
+  'meetingId',
+  'meetingNumber',
+  'neighbourhoodId',
+  'planningApplicationNumber',
+  'reference',
+  'subjectTerms',
+  'termId',
+  'termYear',
+  'wardId',
+] as const;
+
+type _AssertAgendaItemColumnsComplete = Assert<
+  ColumnsMatch<AgendaItem, typeof agendaItemColumns>
+>;
```

Comment:
Does the assert give us additional checking what `satisfies` can do?
```suggestion
] as const satisfies ReadonlyArray<keyof AgendaItem>;
```

Reply:
satisfies will guarantee that all strings in the array are keys of the object, but it _won't_ guarantee that all keys of the object are in the array. so e.g. removing `'wardId'` would error with this gross typescript nonsense i wrote, but not with the nice clean typescript suggestion you made :shrug:. it's a bit overwrought and i'm happy to remove it, i just like types and programming languages so typescript nonsense makes me happy (until it makes me sad)

Reply:
I also love writing elaborate TS types, I think it scratches a similar itch to super compact regexes.
Both are also much more fun to write than to read/debug 😆 

If we wanna have a utility types folder we could add a few of these things and add some explanations to make it more approachable.